### PR TITLE
#39 update code to match updated DB schema

### PR DESCRIPTION
**Pull request by WitchDevelops**
- [x] created a DB schema with new tables
- [x] tested the schema and calls to  `pets_more_info`, `pets_vaccinations`, and `pets_medications`
- [x] fetched those data and displayed in respective tabs
- [x] exported pet data interface to a separate file and used it in pet tabs and child components
- [ ] `pet_tags` will need some more thought (many-to-many relationship, and we'll probably use some library for this)

drawSQL-image-export-2024-06-30

### CODE REVIEW

**Comment by WitchDevelops** [comment]
plz watch the console, there's some error from next/image that idk when it came up or what to do about it - is it also on main? if yes, then what did I mess up?

File: src/pages/user/pets/[pet_id].js [hunk 1]
```diff
@@ -11,29 +11,39 @@ import { PetTabs } from '@/components/PetTabs';
...
-    params: { id: pet.id },
+    params: { pet_id: pet.pet_id },
   }));
 
   return { paths, fallback: false };
 }
 
 export async function getStaticProps({ params }) {
-  const { data: pet, error } = await supabase
+  const { data: pet, error: petError } = await supabase
```
Comment (WitchDevelops) [inline src/pages/user/pets/[pet_id].js:31]: I suppose this DB call could be exported to a separate file

File: src/utils/types/petData.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export interface PetData {
+  pet_id: string;
+  pet_name: string;
+  breed?: string;
+  date_of_birth: Date;
+  species: string;
+  neutered: boolean;
+  sex?: 'female' | 'male';
+  img?: string;
+  pets_more_info?: { pet_bio?: string; microchip_nr?: string };
```
Comment (WitchDevelops) [inline src/utils/types/petData.ts:10]: this prolly can also be extracted into a separate interface

File: src/components/Vaccinations.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { supabase } from '@/services/supabase';
+
+interface VaccinationsProps {
+  pet_id: string;
+}
+
+export const Vaccinations: React.FC<VaccinationsProps> = ({ pet_id }) => {
+  const [vaccinations, setVaccinations] = React.useState<any[]>([]);
```
Comment (WitchDevelops) [inline src/components/Vaccinations.tsx:9]: any, will fix later
(needs an interface)

File: src/components/PetCard.tsx [hunk 4]
```diff
@@ -17,44 +17,44 @@ import { PetCardImage } from '@/components/img/PetCardImage';
...
             >
-              {name}
+              {pet_name}
 
               <TbAlertCircle className="text-danger" />
             </CardTitle>
             <p>{breed}</p>
             <div className="flex justify-between items-center ">
-              {gender ? (
+              {sex ? (
```
Comment (WitchDevelops) [inline src/components/PetCard.tsx:57]: changed coz that is a proper biological term

File: src/components/BasicInfo.tsx [hunk 5]
```diff
@@ -19,24 +21,31 @@ export const BasicInfo = ({ pet }: any) => {
...
-          height={200}
-          width={100}
-          className="object-cover h-[100%] w-[100%]"
-        />
+        {pet.img ? (
+          <Image
+            src={pet.img}
+            alt={pet.pet_name}
+            height={400}
+            width={200}
```
Comment (WitchDevelops) [inline src/components/BasicInfo.tsx:29]: bumped up the image size to fix poor resolution

File: src/components/PetTabs.tsx [hunk 6]
```diff
@@ -24,10 +16,14 @@ export const PetTabs = (props: PetTabsProps) => {
           <TabsTrigger value="appointments">Appointments</TabsTrigger>
         </TabsList>
         <TabsContent value="basicInfo">
-          <BasicInfo pet={props} />
+          <BasicInfo {...pet} />
+        </TabsContent>
+        <TabsContent value="vaccinations">
```
Comment (WitchDevelops) [inline src/components/PetTabs.tsx:21]: would be awesome to add <Suspense> or something to display while fetching the data

**Review by jagodakubicka — CHANGES_REQUESTED** [review]
- AddPetForm not working, as console throws error: `Could not find the 'name' column of 'pets_data' in the schema cache`
 - pet's name not displayed in PetGrid - PetCard
image
 - Cannot enter Pet's Data as id is undefined `http://localhost:3000/user/pets/undefined`

**Review by jagodakubicka — CHANGES_REQUESTED** [review]
- refactor data fetching into try/catch
- clean console.log(pet)

**Review by jagodakubicka — APPROVED** [review]

**Pull request merged**